src/ducks/wallet/lyra.ts

```diff
@@ -0,0 +1,76 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { getPublicKey } from "@stellar/lyra-api";
+
+import {
+  ActionStatus,
+  RejectMessage,
+  WalletInitialState,
+} from "constants/types.d";
+
+export const fetchLyraStellarAddressAction = createAsyncThunk<
+  { publicKey: string },
+  undefined,
+  { rejectValue: RejectMessage }
+>(
+  "walletLyra/fetchLyraStellarAddressAction",
+  async (_, { rejectWithValue }) => {
+    try {
+      const lyraResponse = await getPublicKey();
+
+      if (lyraResponse.error) {
```

Comment:
ah, an error from lyra-api should throw an error - I'll create a ticket to address this